### PR TITLE
Implement HAProxy backend server resource

### LINKED ISSUE #9: M3-02 Implement HAProxy backend server resource
Scope: implement pfsense_haproxy_backend_server as a child of backend with safe parent re-query before mutations.

Acceptance:
- Backend server CRUD/import works against mocks.
- UAT acceptance covers create/update/delete under test backend.
- Parent deletion/order edge cases are tested.

### PR DESCRIPTION
**Pull request by d3vi1**
## Summary

Implements issue #9 / M3-02 for `pfsense_haproxy_backend_server` as a conservative child resource under `pfsense_haproxy_backend`.

- Adds backend server CRUD/import using `backend_name/server_name` as Terraform's stable natural key.
- Re-queries the parent backend before child writes and resolves current pfSense parent/server IDs before mutations.
- Keeps HAProxy reload explicit through `pfsense_haproxy_apply`; no hidden apply behavior is added.
- Adds deterministic mock tests for create, read parent deletion, update order/missing parent, delete order/missing parent, import, and schema boundaries.
- Updates README, schema docs, and the backend example with explicit apply triggers covering backend servers.

Fixes #9.

## Validation

- `terraform fmt -recursive`
- `terraform fmt -check -recursive`
- `go test ./...`
- `make lint`
- `git diff --check`

## Notes

UAT acceptance was not run in this environment. The implementation remains gated by the documented UAT assumptions for `GET /services/haproxy/backend/servers?parent_id=...&name=...` response shape and child ID behavior.

### CODE REVIEW

**Comment by d3vi1** [comment]
@codex, review

**Comment by d3vi1** [comment]
@codex, review

File: examples/haproxy_backend.tf [hunk 1]
```diff
@@ -33,6 +47,15 @@ resource "pfsense_haproxy_apply" "app_backend" {
       monitor_uri         = pfsense_haproxy_backend.app.monitor_uri
       monitor_httpversion = pfsense_haproxy_backend.app.monitor_httpversion
     }))
+    backend_servers = sha1(jsonencode({
+      app01 = {
+        name    = pfsense_haproxy_backend_server.app01.name
+        address = pfsense_haproxy_backend_server.app01.address
+        port    = pfsense_haproxy_backend_server.app01.port
+        status  = pfsense_haproxy_backend_server.app01.status
+        weight  = pfsense_haproxy_backend_server.app01.weight
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply examples/haproxy_backend.tf]: Fixed in 6c17d74: included `ssl` and `sslserververify` in the backend-server apply trigger hashes in both the example and README snippet, so changes to all managed backend-server attributes trigger the explicit apply resource.

File: internal/provider/haproxy_backend_server.go [hunk 2]
```diff
@@ -0,0 +1,772 @@
...
+	return nil
+}
+
+func haproxyBackendServerBackendName(value types.String) (string, error) {
+	name, err := haproxyBackendName(value)
+	if err != nil {
+		return "", fmt.Errorf("backend_name %s", err.Error())
+	}
+	if !haproxyNamePattern.MatchString(name) {
+		return "", fmt.Errorf("backend_name may contain only letters, numbers, dot, hyphen, and underscore")
```
[the remark this answers is not in the document]
Comment (d3vi1) [reply internal/provider/haproxy_backend_server.go:449]: Fixed in 3c767eb on PR #32: `backend_name` for `pfsense_haproxy_backend_server` now delegates to the same backend-name validation used by `pfsense_haproxy_backend`, and that shared backend-name contract now rejects `/` because child resource IDs use `/` as their natural-key separator. The child server `name` keeps its stricter server-name validation. Added regression coverage for both cases.
